### PR TITLE
fix: do not render favorite favStars and filters for anonymous user

### PR DESCRIPTION
**Pull request by trepmag**
### Do not render favorite favStars and filters for charts and dashboards lists for anonymous user

When browsing browsing those lists as anonymous user then 2 issues appears:
1. ~A flash message popup with the text: "There was an error fetching the favorite status: Internal Server Error"~ => fixed in #14287
2. Favorites switches (favStar) and filter are displayed

### Screenshots

Before:
See issue #14070 screenshot.

After:
image

### TEST PLAN

Manually:
1. add some charts and dashbords to be visible by Public role
2. browse as an anonymous user to charts and dashboards lists as anonymous user
3. verify that above described issue (2) doesn't happens

Automatic test:
...

### CODE REVIEW

**Comment by trepmag** [comment]
Last commit 43fe481 for lint issue has 2 problems:
- impose lots of diff hunks
- trigger an exception in charts list: `Unexpected error: Error: A column ID (or string accessor) is required!`

Will digg into this later...

**Comment by amitmiran137** [comment]
Thank you for contributing your first PR!

File: superset-frontend/src/views/CRUD/dashboard/DashboardList.tsx [hunk 1]
```diff
@@ -410,18 +420,22 @@ function DashboardList(props: DashboardListProps) {
...
+            id: 'id',
+            urlDisplay: 'favorite',
+            input: 'select',
+            operator: FilterOperators.dashboardIsFav,
+            unfilteredLabel: t('Any'),
+            selects: [
+              { label: t('Yes'), value: true },
+              { label: t('No'), value: false },
+            ],
+          } as Filter,
```
Comment (suddjian) [inline superset-frontend/src/views/CRUD/dashboard/DashboardList.tsx:436]: Is there any way we can cleanly avoid using `as Filter` here? Ideally the type system would know that this object is compatible without a type assertion.

**Review by suddjian — APPROVED** [review]
One nit, would be nice if it was addressed but it's non-blocking. This LGTM.

**Comment by rusackas** [comment]
@trepmag Thank you for your first PR! I just noticed it's been lingering here a while. Would you like to rebase it and consider @suddjian 's request? We'd love to get this merged - let us know if we can help!

**Comment by trepmag** [comment]
@suddjian, indeed, removing the `as Filter` render a compiler ERROR:
```$
TS2322: Type '{ Header: string; id: string; urlDisplay: string; input: string; operator: FilterOperator; unfilteredLabel: string; selects: { label: string; value: boolean; }[]; }' is not assignable to type 'Filter'.
  Types of property 'input' are incompatible.
    Type 'string' is not assignable to type '"select" | "textarea" | "text" | "search" | "checkbox" | "datetime_range" | undefined'.
```

At this moment I can't figure a way to remove this type casting for this conditional definition...

**Comment by trepmag** [comment]
I rebased manually into one commit a fix for hiding favorite's favStars and filters for char and dashbord lists (the flash message issue was fixed in btw into another issue: #14287).

Although, I notice a related left over issue; the `favstar` table can be altered by anyone by alternating the following curls:
```sh
$ curl "http://localhost:9000/superset/favstar/Dashboard/1/select/"
$ curl "http://localhost:9000/superset/favstar/Dashboard/1/unselect/"
``` 

The `select` operation insert a row as follow:
image

File: superset/views/core.py [hunk 2]
```diff
@@ -1747,6 +1747,8 @@ def favstar(  # pylint: disable=no-self-use
         self, class_name: str, obj_id: int, action: str
     ) -> FlaskResponse:
         """Toggle favorite stars on Slices and Dashboard"""
+        if not g.user.get_id():
+            return json_error_response(f"ERROR: Favstar toggling denied", status=403)
```
Comment (suddjian) [inline superset/views/core.py:1751]: the linter doesn't like that this is an f-string, it should just be a regular string

Comment (trepmag) [reply superset/views/core.py]: thank you, resolved this in 07e6e6c + includes your suggestion for favorites filter cast.

**Comment by suddjian** [comment]
Hi @trepmag, I'm wondering if you were interested in trying out my proposed fix for the Filter type. I'm happy to merge this as is if you prefer, just let me know.

**Comment by trepmag** [comment]
Hi @suddjian, isn't what I did in 07e6e6c? It set `const favoritesFilter: Filter = {...}` as you suggest...

**Comment by suddjian** [comment]
Oh you're absolutely right, I missed that somehow. Thanks!

No need to squash, commits will be squashed automatically when the PR is merged.

**Pull request merged**